[Feat] : 이벤트 상세조회 기능 구현 및 개선

D0ri123 opened the pull request:
### ✒️ 관련 이슈번호

- Close #50 

## 🔑 Key Changes

1. 중복된 내용의 DTO를 사용목적에 맞게 이름을 변경하였습니다. (EventWithProfileDto로 통합)
2. 조회할 이벤트와 그 데이터에서 작성자를 찾기 위해서는 N+1문제가 발생한다는 것을 인지하고, 이를 개선하였습니다.

## 📢 To Reviewers

- None

D0ri123 committed:
feat : 이벤트 상세조회 기능 구현 (#50)

D0ri123 committed:
feat : 새로운 dto를 생성하여 이벤트 상세조회 기능 구현 (#50)

D0ri123 committed:
refactor : 중복 코드 제거 (#50)

D0ri123 committed:
Merge branch 'develop' into feat/#50

D0ri123 committed:
refactor : 쿼리 성능 최적화 및 dto 네이밍 변경 (#50)

soochangoforit commented on src/main/java/dnd/danverse/domain/event/dto/response/EventWithProfileDto.java:
```diff
@@ -34,9 +35,35 @@ public class EventSavedResponseDto {
+   */
+  @Builder
+  public EventWithProfileDto(Event event, ProfileDto profile) {
+    this.profile = profile;
```
API 명세서에는 이벤트 작성 후 응답 값에는 id 필드를 제공하고 있는데 제공하지 않는 이유가 있을까요?? 

전 개인적으로 이벤트에 대한 id 값을 제공하면 이벤트 지원자가 신청하기를 누를 때 이벤트 데이터 안에 있는 id를 통해서 신청할 수 있겠구나 판단했어요

D0ri123 replied:
작성, 조회 responseDTO에 착오가 있었네요! 바로 반영하겠습니다.

soochangoforit commented on src/main/java/dnd/danverse/domain/event/repository/EventRepository.java:
```diff
@@ -10,6 +12,8 @@
-public interface EventRepository extends JpaRepository<Event, Long> , EventFilterCustom{
+public interface EventRepository extends JpaRepository<Event, Long>, EventFilterCustom {
+  @Query("select e from Event e join fetch e.profile where e.id = :eventId")
+  Event findProfileJoinFetch(@Param("eventId") Long eventId);
```
join fetch를 활용하여 2번의 쿼리를 1번의 쿼리문으로 잘 최적화 한 것 같습니다.

하지만, Optional를 활용하지 않아 존재하지 않는 이벤트 ID에 대한 예외 처리가 추가적으로 필요해 보여요, Optional를 활용하면 .orElseThrow를 통해서 예외 처리가 가능해요

저가 만약에 코드를 작성 했다면, EventPureService 에서 Event event = eventRepository.findProfileJoinFetch(eventId).orElseThrow(()-> new EventNotFound(EVENT_NOT_FOUND); 로 하며 EventSearchComplexService에서 PureService에서 받은 Entity를 Dto로 매핑해주는 방향으로 해봤을거 같아요

D0ri123 replied:
말씀해주신 방향대로 다시 구현해보겠습니다😃

soochangoforit commented on src/main/java/dnd/danverse/domain/event/service/EventPureService.java:
```diff
@@ -46,21 +50,40 @@ private Event getEvent(Long eventId) {
+  @Transactional(readOnly = true)
+   public EventWithProfileDto getEventDetail(Long eventId) {
+    Event event = eventRepository.findProfileJoinFetch(eventId);
+    return new EventWithProfileDto(eventId, event, new ProfileDto(event.getProfile()));
```
저희가 추구했던 PureService는 Entity만 응답 하도록 설계를 했는데 수정이 가능할까요?

D0ri123 replied:
넵 바로 수정하겠습니다

D0ri123 committed:
refactor : 코드리뷰 반영한 코드 (#50)

- eventpureservice에서 event엔티티를 바로 반환하도록 변경
- eventrepository에서 optional로 반환받을 수 있도록 변경
- eventwithprofiledto id값 추가

D0ri123 committed:
refactor : 복합 서비스, 순수 서비스 역할 분리 (#50)

D0ri123 committed:
chore : profiledto에서 profilesimpledto로 네이밍 변경 (#50)

soochangoforit reviewed (commented):
저희가 추구했던 복합 Service와 순수 Service에 맞게 잘 분리를 하셨네요 감사합니다 :)
